### PR TITLE
Chore: functional test framework using CodeceptJS

### PR DESCRIPTION
**Pull request by DanDeMicco**
Changes from selenium to [CodeceptJs](https://codecept.io/) which provides a wrapper around [WebdriverIO](https://codecept.io/helpers/WebDriverIO/) to make e2e tests easier. Requires node v8.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @DanDeMicco has signed the CLA. Thanks for the pull request!

**Comment by DanDeMicco** [comment]
A build which passed using this configuration https://travis-ci.org/box/box-content-preview/jobs/337738524

File: codecept.conf.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    typeof SAUCE_USERNAME === 'undefined'
+        ? webDriverIOlocal
+        : Object.assign({}, webDriverIOlocal, {
+            host: 'ondemand.saucelabs.com',
+            port: 80,
+            user: SAUCE_USERNAME,
+            key: SAUCE_ACCESS_KEY,
+            desiredCapabilities: {
+                'tunnel-identifier': TRAVIS_JOB_NUMBER,
+                browserName: 'chrome',
```
Comment (jeremypress) [inline codecept.conf.js]: Break this out as an env variable to support parallelization?

File: .travis.yml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 language: node_js
 node_js:
-- '6'
+- '8'
```
Comment (jeremypress) [inline .travis.yml]: Any repercussions from this?

File: functional-tests/README.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# Running tests locally
```
Comment (jeremypress) [inline functional-tests/README.md]: 💯

File: .travis.yml (hunk 2 above)
Comment (DanDeMicco) [reply .travis.yml]: @jeremypress we will have to update to node 8, which is pretty widely adopted at Box. I have also been using node 8 for the past few days and it has been fine.

**Pull request merged**